inpage/src/WaxStorage.ts

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/require-await */
+import { mapValues } from '@s-libs/micro-dash';
+import z from 'zod';
+
+const typeAndDefault = <T>(type: z.ZodType<T>, default_: T) => ({
+  type,
+  default_,
+});
+
+const schema = {
+  connectedAccounts: typeAndDefault(z.array(z.string()), []),
+};
+
+type Field<T> = {
+  get(): Promise<T>;
+  set(value: T): Promise<void>;
+  clear(): Promise<void>;
+};
+
+export type WaxStorage = {
+  [K in keyof typeof schema]: Field<z.infer<(typeof schema)[K]['type']>>;
+} & {
+  clear(): Promise<void>;
+};
+
+export default function makeLocalWaxStorage() {
+  const fields = mapValues(schema, ({ type, default_ }, key) => ({
```

Comment:
I'm going to try explaining this to make sure I understand how `mapValues` is being used here, please correct me if I'm wrong:

We want to store certain values in local storage such as `connectedAccounts` so we can persist state when refreshing the page or leaving and coming back. We can call `makeLocalWaxStorage` to create an object that can operate on localStorage using the methods `get`, `set` and `clear`. And the point of this is to make dealing with local storage easier and add some type safety with zod?

Reply:
Yep, exactly.